server.js

```diff
@@ -1,38 +1,87 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
+import dotenv from 'dotenv'
 import mongoose from 'mongoose'
+import listEndpoints from 'express-list-endpoints'
+import designers from './data/designers.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+dotenv.config()
 
+// Mongoose set up
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// Mongoose schema to match the keys from the json 
+const Designer = mongoose.model('Designer', {
+  name: {
+    type: String,
+    lowercase: true
+  },
+  surname: {
+    type: String,
+    lowercase: true
+  },
+  wasBorn: Number,
+  memberOfChambreSyndicaleDeLaHauteCouture: Boolean
+})
+
+//  Seeding the database from the json
+if (process.env.RESET_DB) {
+  const seedDB = async () => {
+    await Designer.deleteMany()
+    await designers.forEach(item => {
+      new Designer(item).save()
+    })
+  }
+  seedDB()
+}
+
+// Defining the port the app will run on
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// Middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
+// Routes
+
+// First page API page with possible endpoints
+// https://designers-api.herokuapp.com/
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})  
+
+// Endpoint to get all the designers
+// https://designers-api.herokuapp.com/designers
+app.get('/designers', async (req, res) => {
+  const designers = await Designer.find()
+  res.json(designers)
+})
+
+// Endpoint to get designer by ID
+// E.g https://designers-api.herokuapp.com/designers/609fa52d1dafed06ad8dda19
+app.get('/designers/:designerId', async (req, res) => {
+  const { designerId } = req.params
+  const singleDesigner = await Designer.findById(designerId)
+  res.json(singleDesigner)
+})
+
+// Endpoint to get designer by name
+// E.g https://designers-api.herokuapp.com/designers/name/vivienne
+app.get('/designers/name/:designerName', async (req, res) => {
+  const { designerName } = req.params
+
+  try {
+    const singleDesigner = await Designer.findOne({ name: designerName })
```

Comment:
I think this parameter maybe could be changed to "first name" instead (in the json too) to make it even more clear since for a user! Since it is not possible to use for filtering on the last name (at least when I tried :P )